src/backends/sentry_backend_breakpad.cpp

```diff
@@ -89,17 +97,40 @@ sentry__enforce_breakpad_transport(
     ((sentry_options_t *)options)->transport = transport;
 }
 
+#ifdef SENTRY_PLATFORM_WINDOWS
+static bool
+sentry__breakpad_backend_callback(const wchar_t *breakpad_dump_path,
+    const wchar_t *minidump_id, void *UNUSED(context),
+    EXCEPTION_POINTERS *UNUSED(exinfo), MDRawAssertionInfo *UNUSED(assertion),
+    bool succeeded)
+#else
 static bool
 sentry__breakpad_backend_callback(
     const google_breakpad::MinidumpDescriptor &descriptor,
     void *UNUSED(context), bool succeeded)
+#endif
 {
+#ifndef SENTRY_PLATFORM_WINDOWS
     sentry__page_allocator_enable();
     sentry__enter_signal_handler();
+#endif
 
     const sentry_options_t *options = sentry_get_options();
     sentry__write_crash_marker(options);
-    const char *dump_path = descriptor.path();
+
+    sentry_path_t *dump_path = nullptr;
+#ifdef SENTRY_PLATFORM_WINDOWS
+    dump_path = sentry__path_new(breakpad_dump_path);
+    dump_path = sentry__path_join_wstr(dump_path, minidump_id);
+    dump_path = sentry__path_append_str(dump_path, ".dmp");
+#else
+    dump_path = sentry__path_new(descriptor.path());
+#endif
+
+    // since we can’t use HTTP in crash handlers, we will swap out the
+    // transport here to one that serializes the envelope to disk
+    sentry_transport_t *transport_original = options->transport;
+    sentry__enforce_disk_transport();
```

Comment:
👍 Why didn’t I think of that back when I did this. Seems so obvious in hindsight.